docs/core/integration-quality-scale/rules/iqs003.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "IQS003 - Don't allow the same device or service to be able to be set up twice"
+---
+
+IQS003 is as follows:
+
+Don't allow the same device or service to be able to be set up twice
+
+## Reasoning
+
+Since integrations are easy to setup with the UI, it could occur that the user accidentally sets up the same device or service twice.
+This can lead to duplicated devices and entities with colliding unique identifiers, which has all kinds of negative side effects.
+Making sure a config entry is uniquely identifiable is also important for any discovery flows, as otherwise they would discover devices that already have been setup.
+
+Because we want to prevent this from happening, we need to make sure that the user can only set up a device or service once.
```

Comment:
```suggestion
Since integrations are easy to set up with the UI, the user could accidentally set up the same device or service twice.
This can lead to duplicated devices and entities with unique identifiers colliding, which has negative side effects.
Any discovery flow must also ensure that a config entry is uniquely identifiable, as otherwise, it would discover devices already set up.

To prevent this, we need to ensure that the user can only set up a device or service once.
```